### PR TITLE
Epoch to Validation Pass

### PR DESCRIPTION
**Pull request by pruksmhc**
Changes: 
In some places in the code, we refer to validation passes as epochs. Changing epoch -> val_pass 
In some places we refer steps to "passes, so n_pass -> n_steps

### CODE REVIEW

File: src/trainer.py [hunk 1]
```diff
@@ -522,8 +522,8 @@ def train(
                         "load_model=1 but there is not checkpoint. Starting training without restoring from a checkpoint."
                     )
                 else:
-                    n_pass, should_stop = self._restore_checkpoint(phase, tasks)
-                    log.info("Loaded model from checkpoint. Starting at pass %d.", n_pass)
+                    n_steps, should_stop = self._restore_checkpoint(phase, tasks)
+                    log.info("Loaded model from checkpoint. Starting at pass %d.", n_steps)
```
Comment (sleepinyourhat) [inline src/trainer.py]: 'Starting at step %d', assuming this really is a step?

File: src/trainer.py [hunk 2]
```diff
@@ -510,7 +510,7 @@ def train(
         self._g_scheduler = g_scheduler
 
         # define these here b/c they might get overridden on load
-        n_pass, should_stop = 0, False
+        n_steps, should_stop = 0, False
```
Comment (sleepinyourhat) [inline src/trainer.py]: => 'step'?

File: src/trainer.py [hunk 3]
```diff
@@ -713,13 +713,13 @@ def clip_function(grad):
 
                 if should_save:
                     self._save_checkpoint(
-                        {"pass": n_pass, "epoch": n_val, "should_stop": should_stop},
+                        {"steps": n_steps, "validation_pass": n_val, "should_stop": should_stop},
```
Comment (sleepinyourhat) [inline src/trainer.py]: steps => step

File: src/utils/utils.py [hunk 4]
```diff
@@ -52,20 +52,19 @@ def check_for_previous_checkpoints(serialization_dir, tasks, phase, load_model):
     ---------------------
     ckpt_directory: None or str, name of directory that checkpoints are in
     with regards to the run directory. 
-    max_epoch: int, -1 if not found.
+    val_passh: int, -1 if not found.
```
Comment (sleepinyourhat) [inline src/utils/utils.py]: => val_pass?

File: tests/test_checkpointing.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def build_trainer_params(args, task_names, phase="pretrain"):
         "val_data_limit": 10,
         "d_hid": 1024,
         "min_lr": 0.000,
-        "max_vals": 4,
+        "max_epochs": 4,
```
Comment (sleepinyourhat) [inline tests/test_checkpointing.py]: Why this switch?

File: tests/test_checkpointing.py [hunk 6]
```diff
@@ -44,7 +44,7 @@ def build_trainer_params(args, task_names, phase="pretrain"):
         "max_grad_norm": 5.0,
         "val_interval": 500,
         "keep_all_checkpoints": 0,
-        "max_epochs": 2,
+        "max_vals": 2,
```
Comment (sleepinyourhat) [inline tests/test_checkpointing.py]: Why this switch?

Comment (sleepinyourhat) [reply tests/test_checkpointing.py]: max_epochs is still a valid config option, IIRC, which refers to real epochs.

File: tests/test_checkpointing.py (hunk 5 above)
Comment (pruksmhc) [reply tests/test_checkpointing.py]: Oh this was a copy-paste-fix error

**Pull request merged**